packages/eslint-plugin/tests/rules/strict-boolean-expressions.test.ts

```diff
@@ -1303,6 +1439,7 @@ if (((Boolean('')) && {}) || (foo && void 0)) { }
         declare const x: string; if (x) {}
         (x: string) => (!x);
         <T extends string>(x: T) => x ? 1 : 0;
+        ['foo', 'bar'].filter(x => x);
```

Comment:
[Testing] Blurgh, `batchedSingleLineTests`. That old deprecated thing. Let's not add more uses of it in this PR.

If there's anything you want to specifically test in here that isn't tested elsewhere, please add it as a standalone test.